update subclasses of has input / has output

sfluegel05 opened the pull request:
closes #618 

> * replace `has numerical input` by `has information input`, defined as _p has information input c iff: p has input c, and c is an information content entity_ (analogous to the definition of `has physical input`).
> * add missing domains and ranges to the children of `has input` to further distinguish them
> * make `has constraint` a subproperty of `has information input`
> * do the same for `has ... output`

jannahastings reviewed (changes requested):
I'm afraid we have inconsistent classes now due to the domain of has-physical-output being set to 'process'. All the power generating classes are subclasses of, or have as part a class that is subclass of, has-physical-output some energy. But they are classified as continuants.

sfluegel05 commented:
That's bad. I don't see an easy solution to this problem since `has participant` and its subclasses did always have the domain `process` in the definition, the only thing I changed was making it explicit in an axiom.
For now I removed the conflicting axioms so we can merge the rest of the PR and solve the inconsistency-problem in a new issue.

jannahastings commented:
> That's bad. I don't see an easy solution to this problem since `has participant` and its subclasses did always have the domain `process` in the definition, the only thing I changed was making it explicit in an axiom.
> For now I removed the conflicting axioms so we can merge the rest of the PR and solve the inconsistency-problem in a new issue.

Agreed.
